python/ml4ir/applications/ranking/tests/test_metrics.py

```diff
@@ -232,4 +306,50 @@ def test_generate_stat_sig_based_metrics(self):
         stat_sig = metrics_helper.generate_stat_sig_based_metrics(df, metric, str(group_keys), metrics_dict)
         assert metrics_dict["stat_sig_" + metric + "_improved_groups"] == 1
         assert metrics_dict["stat_sig_" + metric + "_degraded_groups"] == 1
-        assert metrics_dict["stat_sig_" + metric + "_group_improv_perc"] == 0
\ No newline at end of file
+        assert metrics_dict["stat_sig_" + metric + "_group_improv_perc"] == 0
+
+    def test_add_top_graded_relevance_record_column(self):
+        # Create a sample DataFrame
+        data = {
+            'query_id': [1, 1, 2, 2, 3, 4, 4],
+            'relevance_score': [0.8, 0.5, 0.6, 0.9, 0.7, 0, 0]
+        }
+        df = pd.DataFrame(data)
+
+        # Call the function
+        new_col_name = 'top_relevance'
+        df_with_new_column = metrics_helper.add_top_graded_relevance_record_column(df, 'query_id', 'relevance_score', new_col_name)
+
+        # Verify the new column is added
+        self.assertTrue(new_col_name in df_with_new_column.columns)
+
+        # Verify the values in the new column
+        expected_values = np.array([1.0, 0.0, 0.0, 1.0, 1.0])
+        np.testing.assert_array_equal(df_with_new_column[new_col_name].values, expected_values)
+
+        # Verify rows with no relevance scores are removed
+        self.assertEqual(len(df_with_new_column['query_id'].unique()), 3)
+
+    def test_compute_NDCG(self):
+        data = {
+            'query_id': [1, 1, 1, 2, 2, 2],
+            'y_true': [3.0, 2.0, 1.0, 1.0, 2.0, 3.0],
+            'y_pred': [0.3, 0.2, 0.1, 10, 20, 30]
+        }
+        expected_values = np.array([1.0, 1.0, 1.0, 1.0, 1.0, 1.0])
+        self.comput_and_assert_NDCG(data, expected_values)
+
+        data = {
+            'query_id': [1, 1, 1, 2, 2, 2],
+            'y_true': [3.0, 2.0, 1.0, 1.0, 2.0, 3.0],
+            'y_pred': [0.3, 0.2, 0.1, 30, 20, 10]
+        }
+        expected_values = np.array([1.0, 1.0, 1.0, 0.789998, 0.789998, 0.789998])
+        self.comput_and_assert_NDCG(data, expected_values)
```

Comment:
@mohazahran This should be a separate unit test case.

Reply:
done